tests/core/config/bdk_config_test.py

```diff
@@ -24,3 +27,43 @@ def test_update_certificate(simple_config_path):
     certificate_path = get_resource_filepath("cert/certificate.cert", as_text=True)
     config.bot.certificate.path = certificate_path
     assert config.bot.certificate._path == certificate_path
+
+
+def test_retry_configuration():
+    config_path = get_config_resource_filepath("retry_config.yaml")
+    config = BdkConfigLoader.load_from_file(config_path)
+
+    assert config.retry.max_attempts == 2
+    assert config.retry.initial_interval.total_seconds() == 1.0
+    assert config.retry.multiplier == 3
+    assert config.retry.max_interval.total_seconds() == 2.0
+
+
+def test_default_retry_configuration():
+    config_path = get_config_resource_filepath("config.yaml")
+    config = BdkConfigLoader.load_from_file(config_path)
+
+    assert config.retry.max_attempts == BdkRetryConfig.DEFAULT_MAX_ATTEMPTS
+    assert config.retry.initial_interval.total_seconds() == timedelta(
```

Comment:
Why testing with `total_seconds()` in each side of the `==` test ? Is that not working with `assert config.retry.initial_interval == timedelta(milliseconds=BdkRetryConfig.DEFAULT_INITIAL_INTERVAL)`?